### PR TITLE
Implement stable order for `-node-containerd-registry-mirrors` flags in OSM Deployment

### PR DESCRIPTION
**Pull request by embik**
**What this PR does / why we need it**:
This issue was reported by @dharapvj. The `node-containerd-registry-mirrors` flags set on the OSM deployment are generated from iterating over a map. Unfortunately, Go maps are famously random in iteration, so every run this can generate a different list of flags, which forces constant re-deployment.

This PR implements sorting over the map keys so a stable order is maintained when looping. I could have also sorted the flags after they were generated, but this felt like the "proper" way to do it (also it might have not retained priority listing for the endpoints, if such a thing exists).

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind bug

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Stop constantly re-deploying operating-system-manager when registry mirrors are configured
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: pkg/resources/operatingsystemmanager/deployment.go [hunk 1]
```diff
@@ -333,12 +334,21 @@ func getEnvVars(data operatingSystemManagerData) ([]corev1.EnvVar, error) {
...
+	// for stable order.
+	for registry := range crid.Registries {
+		registries = append(registries, registry)
+	}
+
+	slices.Sort(registries)
+
+	for _, registry := range registries {
+		for _, endpoint := range crid.Registries[registry].Mirrors {
+			flags = append(flags, fmt.Sprintf("-node-containerd-registry-mirrors=%s=%s", registry, endpoint))
```
Comment (xrstf) [inline pkg/resources/operatingsystemmanager/deployment.go:351]: What about simply sorting `flags` after this loop?

**Comment by embik** [comment]
@xrstf @ahmedwaleedmalik can one of you please take a look? It would be great if we could include this in our upcoming patch releases next week.

**Review by xrstf — APPROVED** [review]
I would have used a string.Set or simply sorted the flags, but alas..

/approve

**Pull request merged**